### ISSUE TITLE
Stacktrace_unittest fails on Alpine 3.12 docker

### ISSUE THREAD

**Issue by konstantia86** [gperftools/gperftools#1282]
Hi everyone,

I am trying to port gperftools (2.9.1) in Alpine 3.12 via docker. While using libunwind and libunwind-dev, the stacktrace_unittest and the heap-checker_unittest were failing. As soon as I recompiled without libunwind but with musl and libgcc, only the stacktrace_unittest is failing. Unfortunately, due to docker restrictions (breaking the operation while make check fails), I'm not able to access the test-log file. Any ideas on how to approach the troubleshooting?

The packages I am using (system-wide, I have more apps to port) are:  bash ca-certificates py3-pip python3-dev python3 tzdata rsync p7zip openssh openssh-client bind-tools curl py-pip libffi-dev openssl-dev libc-dev make openjdk8 ethtool net-tools screen sudo syslog-ng tcpdump gdb sqlite sqlite-dev stunnel python2 python2-dev cmake make gcc g++ flex bison libpcap-dev libressl-dev swig zlib-dev mini-sendmail krb5-dev open-vm-tools autoconf git netcat-openbsd libmaxminddb libmaxminddb-dev automake libgcc libtool musl

Thanks in advance for your help.

**Comment by alk** [comment]
Yes, seeing this with latest code too. You can work around by using frame pointers with TCMALLOC_STACKTRACE_METHOD=generic_fp (but then app needs to be built with frame pointers).

Another alternative with libunwind is

https://github.com/gperftools/gperftools/blob/2b5743ffa60affcdc796dcf4bf6c4f5fff14c4de/src/stacktrace_libunwind-inl.h#L63

I.e. set this define and libunwind won't need to step through signal handle frame.

I would like to look at this too, but unsure when I'll be able to get this. First guess is musl lacks unwind info declarations in their asm code for signal trampolines (which should be straightforward for someone to contribute btw).
